pr_agent/git_providers/github_provider.py

```diff
@@ -187,24 +184,19 @@ def publish_code_suggestions(self, code_suggestions: list):
             else:  # API is different for single line comments
                 post_parameters = {
                     "body": body,
-                    "commit_id": self.last_commit_id._identity,
                     "path": relevant_file,
                     "line": relevant_lines_start,
                     "side": "RIGHT",
                 }
+            post_parameters_list.append(post_parameters)
 
-            try:
-                headers, data = self.pr._requester.requestJsonAndCheck(
-                    "POST", f"{self.pr.url}/comments", input=post_parameters
-                )
-                github.PullRequestComment.PullRequestComment(
-                    self.pr._requester, headers, data, completed=True
-                )
-                return True
-            except Exception as e:
-                if settings.config.verbosity_level >= 2:
-                    logging.error(f"Failed to publish code suggestion, error: {e}")
-                return False
+        try:
+            self.pr.create_review(commit=self.last_commit_id, comments=post_parameters_list)
+            return True
+        except Exception as e:
+            if settings.config.verbosity_level >= 2:
+                logging.error(f"Failed to publish code suggestion, error: {e}")
+            return False
```

Comment:
**Suggestion:** Consider handling specific exceptions instead of a general exception to avoid suppressing unexpected issues.
```suggestion
        try:
            self.pr.create_review(commit=self.last_commit_id, comments=post_parameters_list)
            return True
        except GithubException as e:
            if settings.config.verbosity_level >= 2:
                logging.error(f"Failed to publish code suggestion, error: {e}")
            return False
```